nmt.go

```diff
@@ -115,6 +119,7 @@ func New(h hash.Hash, setters ...Option) *NamespacedMerkleTree {
 		leaves:          make([][]byte, 0, opts.InitialCapacity),
 		leafHashes:      make([][]byte, 0, opts.InitialCapacity),
 		namespaceRanges: make(map[string]leafRange),
+		// TODO [Me] Shouldn't minNID be populated by `0x00`?
```

Comment:
I see, then, does it mean that the `minID` is actually equal to `0xFF` which corresponds to the Max namespace ID?

Reply:
> That said, I think we can get rid of these two fields entirely!

Noted!

Reply:
Update: I just double checked the code, and figured these two fields represent the range of the namespace IDs of the data stored by a neamspace tree, and they are accessed in here:
https://github.com/celestiaorg/nmt/blob/ac87f1cd1331b2a6b72b14ab427cffeddccaa154/nmt.go#L174
This means, as it is, we cannot remove these two fields. 
However, we could retrieve the same information i.e., the namespaceID range by looking at the tree [`rawRoot`](https://github.com/celestiaorg/nmt/blob/ac87f1cd1331b2a6b72b14ab427cffeddccaa154/nmt.go#L95) field (as it is supposed to embody the max and min namespace IDs of the entire tree). However, that requires a bit of refactoring so that the root gets calculated after each `Push` operation (Which is not currently the case).

Reply:
Yes, removing these fields will need some refactoring. I did not think it through beyond the fact that the info about min/max namespaces also materializes in the root.

Reply:
> I see, then, does it mean that the minID is actually equal to 0xFF which corresponds to the Max namespace ID?

No, this is only to initialize the minID such that the if-block yields true for any namespace smaller than the max one (and hence the minID gets updated): https://github.com/celestiaorg/nmt/blob/ac87f1cd1331b2a6b72b14ab427cffeddccaa154/nmt.go#L174